### PR TITLE
Update calibration program to latest version, from OpenCV 4

### PR DESCRIPTION
**Pull request by evan3334**
This pull request updates the calibration sample program to the latest version that is available in OpenCV 4. Notable changes include the use of the "releasing-object" method of camera calibration, which is more accurate for less precise calibration targets (like the printed paper targets we will most likely be using). We probably ought to use this when we calibrate the cameras on the rover. I'll update the documentation on camera calibration to include info on this.

Note that this new program uses functions that have been introduced in OpenCV 4, so as such OpenCV 4 will be *required* to build it. I don't actually think it will affect the entire project, you have to explicitly specify that you want to build the calibration program so people using OpenCV 3 should still be fine for the time being - however, we probably want to have everyone update at some point.

### CODE REVIEW

File: src/ar/calibration.cpp [hunk 1]
```diff
@@ -106,19 +103,25 @@ static void help(char** argv)
...
+        "     [-dt=<distance>]         # actual distance between top-left and top-right corners of\n"
+        "                              # the calibration grid. If this parameter is specified, a more\n"
+        "                              # accurate calibration method will be used which may be better\n"
+        "                              # with inaccurate, roughly planar target.\n"
         "     [input_data]             # input data, one of the following:\n"
         "                              #  - text file with a list of the images of the board\n"
         "                              #    the text file can be generated with imagelist_creator\n"
         "                              #  - name of video file with a video of the board\n"
         "                              # if input_data not specified, a live view from the camera is used\n"
-        "\n", argv[0] );
```
Comment (nhatch) [inline src/ar/calibration.cpp:121]: Why this change? Seems like a fine practice to use argv for this, in case we ever change the filename or name of the executable

File: src/ar/calibration.cpp [hunk 2]
```diff
@@ -194,16 +199,32 @@ static bool runCalibration( vector<vector<Point2f> > imagePoints,
...
     objectPoints.resize(imagePoints.size(),objectPoints[0]);
 
-    double rms = calibrateCamera(objectPoints, imagePoints, imageSize, cameraMatrix,
-                    distCoeffs, rvecs, tvecs, flags|CALIB_FIX_K4|CALIB_FIX_K5);
-                    ///*|CALIB_FIX_K3*/|CALIB_FIX_K4|CALIB_FIX_K5);
+    double rms;
+    int iFixedPoint = -1;
+    if (release_object)
+        iFixedPoint = boardSize.width - 1;
+    rms = calibrateCameraRO(objectPoints, imagePoints, imageSize, iFixedPoint,
```
Comment (nhatch) [inline src/ar/calibration.cpp:211]: Ok so I guess this is the new OpenCV4 API call? (Not actually a question, just sort of trying to prove I'm paying attention to what's going on.)

Small suggestion: instead of `using namespace cv` above, just use `cv::calibrateCameraRO` here. That makes it clear that this is an OpenCV method. We don't have to do that in this PR, but just as a suggestion for the future. "using namespace" usually makes code a bit less clear to unfamiliar readers.

**Review by nhatch — APPROVED** [review]
For posterity, here's the documentation on camera calibration: https://app.clickup.com/2200143/v/dc/234jf-1778/234jf-3969

Looks good to me.

File: src/ar/calibration.cpp (hunk 1 above)
Comment (evan3334) [reply src/ar/calibration.cpp:121]: No idea, I just copied in their sample program basically. Not sure why they made that change but if it doesn't make sense we can just exclude it

File: src/ar/calibration.cpp (hunk 2 above)
Comment (evan3334) [reply src/ar/calibration.cpp:211]: Yeah, that makes sense. OpenCV seems pretty eager to use `using namespace` in their sample code which isn't great practice, I agree. I can just modify that now.

**Comment by evan3334** [comment]
Made the recommended changes, so I think I'm just going to go ahead and merge this.
